packages/toggle/src/Toggle.tsx

```diff
@@ -24,431 +25,415 @@ export const Mode = {
-  `,
-);
-
-const getStatefulStyles = (states: StateForStyle) => ({
-  slider: getSliderStyles(states),
-  groove: getGrooveStyles(states),
-  container: getContainerStyles(states),
-});
+    onLabel: css`
+      color: ${uiColors.white};
+      text-shadow: 0 0 2px ${uiColors.blue.base};
+    `,
+  },
+} as const;
 
 interface BaseToggleProps {
+  /**
+   * Sets the size of the toggle.
+   *
+   * default: `'default'`
+   */
   size?: Size;
+
+  /**
+   * Determines if the Toggle will render the dark mode styles.
+   *
+   * default: `false`
+   */
   darkMode?: boolean;
-}
 
-type ToggleProps = BaseToggleProps &
-  Omit<HTMLElementProps<'input', never>, keyof BaseToggleProps>;
+  /**
+   * Sets the checked state of the Toggle.
+   */
+  checked?: boolean;
+
+  /**
+   * Disables the Toggle.
+   *
+   * default: `false`
+   */
+  disabled?: boolean;
+
+  /**
+   * `onChange` fires when the internally-managed `checked` state of the component is updated. Receives the updated checked state of the toggle as its first argument, and the associated mouse event as the second.
+   */
+  onChange?: (
+    checked: boolean,
+    mouseEvent: React.MouseEvent<HTMLButtonElement>,
+  ) => void;
+
+  /**
+   * The event handler function for the 'onclick' event. Receives the associated `event` object as the first argument.
+   */
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+
+  /**
+   * Adds a className to the outermost element.
+   */
+  className?: string;
+}
 
-const idAllocator = IdAllocator.create('toggle');
+type ToggleProps = Either<
+  BaseToggleProps &
+    Omit<HTMLElementProps<'button', never>, keyof BaseToggleProps>,
+  'aria-label' | 'aria-labelledby'
+>;
 
 function Toggle({
-  name,
   className,
   size = Size.Default,
   darkMode = false,
   disabled = false,
-  onChange: onChangeProp = () => {},
+  onChange: onChangeProp,
+  onClick: onClickProp,
   checked: checkedProp,
-  id: idProp,
   ...rest
 }: ToggleProps) {
-  const [checked, setChecked] = useState(false);
-  const toggleId = useMemo(() => idProp ?? idAllocator.generate(), [idProp]);
-  const normalizedChecked = checkedProp || checked;
+  validateAriaLabelProps(rest, Toggle.displayName);
 
-  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    if (onChangeProp) {
-      onChangeProp(e);
-    }
+  const [buttonElement, setButtonElement] = useState<HTMLButtonElement | null>(null);
+  const [checked, setChecked] = useState(checkedProp ?? false);
+  const normalizedChecked = checkedProp ?? checked;
 
-    if (checkedProp == null) {
-      setChecked(e.target.checked);
-    }
-  };
+  const onClick: React.MouseEventHandler<HTMLButtonElement> = useCallback(
+    e => {
+      onClickProp?.(e);
 
-  const statefulStyles = getStatefulStyles({
-    disabled,
-    size,
-    checked: normalizedChecked,
-    mode: darkMode ? Mode.Dark : Mode.Light,
-  });
+      setChecked(curr => {
+        const updatedState = !(checkedProp ?? curr);
```

Comment:
@brookescarlett @lazytype I wanted to call out this change I made to make sure it makes sense.

When you have a controlled checkbox and are responding to its onChange events, I think you get the value that it should be updated to regardless of whether it's controlled or not. Do y'all agree that this makes more sense to pass to the user's onChange callback than simply `!curr`?

Reply:
Taking a closer look, I don't think that it makes sense to break this out into two separate props. They're being triggered by the same event, and there's no difference in how/when they're being called, as the only check is for the respective function's existence. I think we should continue with just the `onChange` prop and passing the event up to the consuming application (as is consistent with most of our form-like elements). If a user wants to use that information to toggle internal state, they are able to do so as-is.

Reply:
The reason I included both is mostly to mirror the feature set of using a checkbox. If we don't call onClick in here,  onClick won't work at all, and that's not really okay. If we don't call onChange here, that won't work either since it's non-standard for buttons.

Reply:
I've been doing some research and it seems to me that `onChange` really is the conventional event handler when it comes to responding to Toggle/Switch components, so I am inclined to say that we follow-suit, and leave the code you have here as-is, but just not explicitly document the onClick prop. That way, onClick won't break anything, but also isn't what we are recommending in terms of how to use the component.

The only drawback to this approach is that I don't love that the type of event that the onChange would expect to receive isn't the type of event that it actually will receive (expect: React.ChangeEvent, receive: React.MouseEvent). Not sure this needs to be of the utmost concern when making this decision, but it is something I thought I'd raise.

Reply:
If it means we're leaving the code here as-is, I don't feel strongly about removing the documentation.

I agree with your feelings about passing a mouse event to a change event handler, but the alternatives are constructing an equivalent of a change event, or not providing one at all. Both seem like worse alternatives than the client receiving the event that is causing an onChange event to fire.